### PR TITLE
DateMath: Added ago/from triggers

### PR DESCRIPTION
**Pull request by W25**
Solved https://github.com/duckduckgo/zeroclickinfo-goodies/issues/879

### CODE REVIEW

File: lib/DDG/Goodie/DateMath.pm [hunk 1]
```diff
@@ -24,8 +24,25 @@ my $datestring_regex = datestring_regex();
 
 handle query_lc => sub {
     my $query = $_;
-    return unless $query =~ qr!^($datestring_regex)\s+(plus|\+|\-|minus)\s+(\d+|[a-z\s-]+)\s+((?:day|week|month|year)s?)$!;
```
Comment (mintsoft) [inline lib/DDG/Goodie/DateMath.pm]: @W25 apologies for the delay; it certainly reads a bit easier to my eyes

**Comment by mintsoft** [comment]
@MrChrisW @moollaza @jagtalon anything to add?

**Comment by jagtalon** [comment]
@mintsoft it looks good to me. You've had a lot of experience with the dates so I defer to your experience here. :)

**Comment by mintsoft** [comment]
@W25 This has merge conflicts now, can you rebase whilst I give it a final checkover?

**Comment by W25** [comment]
@mintsoft Thank you, I merged it.

File: lib/DDG/Goodie/DateMath.pm [hunk 2]
```diff
@@ -6,13 +6,13 @@ with 'DDG::GoodieRole::Dates';
 use DateTime::Duration;
 use Lingua::EN::Numericalize;
 
-triggers any => qw( plus minus + - );
+triggers any => qw( plus minus + - date from ago day week month year days weeks months years);
```
Comment (moollaza) [inline lib/DDG/Goodie/DateMath.pm]: I'm not sure "from" and "ago" are necessary. I think any query that uses those will include one of the time units you have as triggers? (i.e. day(s), week(s), month(s), year(s))

Comment (moollaza) [inline lib/DDG/Goodie/DateMath.pm]: The thing to remember is that adding these as triggers, will cause this code to run for **every** query that has "from" or "ago" in it, which I'm sure the majority of those are quite unlikely to be the kinds of queries we want this IA to deal with

Comment (W25) [inline lib/DDG/Goodie/DateMath.pm]: @moollaza Thank you, I removed the triggers.

**Comment by mintsoft** [comment]
@W25 Thanks for the fixes; apologies for the delays around this I just wanted to give it a good check-through as the dates role is pretty core to several goodies

:cake:

**Pull request merged**